crates/services/gas_price_service/src/fuel_gas_price_updater/tests.rs

```diff
@@ -13,7 +13,7 @@ struct FakeL2BlockSource {
 
 #[async_trait::async_trait]
 impl L2BlockSource for FakeL2BlockSource {
-    async fn get_l2_block(&self, _height: BlockHeight) -> Result<BlockInfo> {
+    async fn get_l2_block(&mut self, _height: BlockHeight) -> Result<BlockInfo> {
         let block = self.l2_block.lock().await.recv().await.unwrap();
```

Comment:
Do we need to use `lock` now if it is `&mut self`?

Reply:
The lock wasn't becasue this wasn't `mut`, it's because we need to access the value in another thread in the test.